library/std/src/os/unix/net/ancillary.rs

```diff
@@ -407,7 +449,13 @@ pub struct SocketAncillary<'a> {
 }
 
 impl<'a> SocketAncillary<'a> {
-    /// Create an ancillary data with the given buffer.
+    /// Create a struct for accessing an ancillary data packet in `buffer`.
+    /// This ancillary data packet will be empty.
+    /// The result won't own the ancillary data and will read them from
+    /// and write them into `buffer`. Obviously, this buffer isn't extensible.
```

Comment:
Since it is obvious it could be rephrased as

> Since the buffer is not extensible it can instead be sized in advance by calculating the aggregate size of all expected control messages, (see ...